ui/de-lib/src/main/java/org/iplantc/de/diskResource/client/presenters/metadata/MetadataPresenterImpl.java

```diff
@@ -89,26 +155,35 @@ public void onFailure(Throwable caught) {
             }
         };
 
-        drService.setDiskResourceMetaData(resource,
-                                          view.getMetadataTemplate(),
-                                          view.getAvus(),
-                                          batchAvuCallback);
+        DiskResourceUserMetadata umd =
+                AutoBeanCodex.decode(autoBeanFactory, DiskResourceUserMetadata.class, "{}").as();
+        umd.setAvus(view.getUserMetadata());
+        drService.setDiskResourceMetaData(resource, umd, view.getAvus(), batchAvuCallback);
```

Comment:
Since we have `org.iplantc.de.client.models.avu.AvuList` now, `DiskResourceMetadataList` can extend `AvuList`, and this `setDiskResourceMetaData` method can take `DiskResourceMetadataList` as a parameter instead of the separate AVU lists.

Also, in order to preserve metadata AVU `created_on` and `created_by` fields, the UI needs to send the original AVU ID in the set request. Otherwise, the service will delete all metadata AVUs and recreate them from the request (which means the `id`, `created_on`, and `created_by` fields are reset in every request). Extending `org.iplantc.de.client.models.avu.AvuList` could allow the UI to preserve the `org.iplantc.de.client.models.avu.Avu` IDs for disk resource metadata.